<!--This is the launch file for launch a single flir_gige nodelet-->
<launch>
    <!--Specifies whether to run nodelet as plugin or standalone-->
    <arg name="plugin" default="false"/>
    <arg name="proc" default="false"/>
    <!--Controls output tag of the node-->
    <arg name="disp" default="screen"/>
    <!--Camera settings-->
    <arg name="ip_address" default="192.168.0.11"/>
    <arg name="frame_id" default="flir_a5"/>
    <arg name="fps" default="20.0"/>
    <arg name="color" default="false"/>
    <arg name="bit" default="2"/>
    <arg name="calib_file" default="$(arg frame_id).yml"/>
    <arg name="calib_url" default="package://flir_gige/calib/$(arg calib_file)"/>

    <!--run as standalone-->
    <node unless="$(arg plugin)" pkg="nodelet" type="nodelet" name="flir_gige"
        args="standalone flir_gige/FlirNodelet" output="$(arg disp)">
        <param name="ip_address" type="string" value="$(arg ip_address)"/>
        <param name="frame_id" type="string" value="$(arg frame_id)"/>
        <param name="fps" type="double" value="$(arg fps)"/>
        <param name="color" type="bool" value="$(arg color)"/>
        <param name="bit" type="int" value="$(arg bit)"/>
        <param name="calib_url" type="string" value="$(arg calib_url)"/>
    </node>

    <!--run as plugin-->
    <group if="$(arg plugin)">
        <arg name="manager" default="flir_gige_nodelet"/>

        <node pkg="nodelet" type="nodelet"
            name="$(arg manager)" args="manager" output="$(arg disp)">
        </node>

        <node pkg="nodelet" type="nodelet" name="flir_gige"
            args="load flir_gige/FlirNodelet $(arg manager)"
            output="$(arg disp)">
            <param name="ip_address" type="string" value="$(arg ip_address)"/>
            <param name="frame_id" type="string" value="$(arg frame_id)"/>
            <param name="fps" type="double" value="$(arg fps)"/>
            <param name="color" type="bool" value="$(arg color)"/>
            <param name="bit" type="int" value="$(arg bit)"/>
            <param name="calib_url" type="string" value="$(arg calib_url)"/>
        </node>

        <node if="$(arg proc)" pkg="nodelet" type="nodelet" name="thermal_proc"
            args="load flir_gige/ThermalProcNodelet $(arg manager)"
            ns="flir_gige"
            output="$(arg disp)">
            <!-- <remap from="~image_raw" to="/flir_gige/image_raw"/> -->
            <!-- <remap from="~camera_info" to="/flir_gige/camera_info"/> -->
        </node>
    </group>
</launch>
